src/quicklist.c

```diff
@@ -924,9 +924,9 @@ REDIS_STATIC void _quicklistInsert(quicklist *quicklist, quicklistEntry *entry,
         new_node->count++;
         quicklistNodeUpdateSz(new_node);
         quicklistRecompressOnly(quicklist, new_node);
-    } else if (full && ((at_tail && node->next && full_next && after) ||
-                        (at_head && node->prev && full_prev && !after))) {
-        /* If we are: full, and our prev/next is full, then:
+    } else if (full && ((at_tail && full_next && after) ||
+                        (at_head && full_prev && !after))) {
+        /* If we are: full, and our prev/next is full (means prev/next is NULL or really full), then:
```

Comment:
```suggestion
        /* If we are: full, and our prev/next is full or NULL, then:
```
My English is pool, so I'm not sure it would be better.